### ISSUE TITLE
Why does the metrics (AP, P, R) and the loss go down to zero in epoch 250?

### ISSUE THREAD

**Issue by dariogonle** [ultralytics/yolov5#5396]
## ❔Question
I've used YOLOv5 several times, but there are some cases where the loss, AP, P and R go down to zero in the middle of the training, as show I the following image.

![results](https://user-images.githubusercontent.com/80393891/139429030-a474c703-ff74-4d32-9245-5773c55c31fb.png)

Why does this happen? It is possible that at computing P or R there is an 0/0 division? I think this may be a possibility since one of my classes has very few samples.

Thank you in advance.

**Comment by glenn-jocher** [comment]
@dariogonle 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

* ✅ **Minimal** – Use as little code as possible that still produces the same problem
* ✅ **Complete** – Provide **all** parts someone else needs to reproduce your problem in the question itself
* ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

In addition to the above requirements, for [Ultralytics](https://ultralytics.com/) to provide assistance your code should be:

* ✅ **Current** – Verify that your code is up-to-date with current GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been resolved by previous commits.
* ✅ **Unmodified** – Your problem must be reproducible without any modifications to the codebase in this repository. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all of the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) and providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by Zengyf-CVer** [comment]
@dariogonle 
There are certain problems with your data set. Check carefully, the sample division of the training set and the validation set, the classification of categories, and the division of instances.
